lib/find-or-add-with-events.js

```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+var findOrAddOne = require('../helpers/find-or-add-one')
+var findOrAddMany = require('../helpers/find-or-add-many')
+
+module.exports = findOrAdd
+
+/**
+ * tries to find object in local database, otherwise creates new one
+ * with passed properties.
+ *
+ * @param  {String|Object} idOrObject   id or object with `.id` property
```

Comment:
we should either add `@param {Object} state` here, or leave out the jsdoc comments altogether. I think I'd go with the latter